### PR TITLE
dogehouse: adds deafen and app download page

### LINKED ISSUE #2043: Mute room
If you want to quickly mute to maybe talk to somebody on a call it would be pretty nice to have an option to mute the room instead of having to mute the tab.

### LINKED ISSUE #2187: Make the avatars more accessible to screen readers
Currently all avatars use an `alt` text of "avatar" which isn't screen reader friendly when having multiple avatars on screen (e.g. voice chat page)

Maybe make the `alt` text more "personalized". Like "Ben's avatar" or "John's avatar"

https://github.com/benawad/dogehouse/blob/611879da2bff5a172f880ac5ee0da82a383b08a7/kibbeh/src/ui/UserAvatar/SingleUser.tsx#L88-L90

### PR DESCRIPTION
**Pull request by amitojsingh366**
- adds deafen
- adds `/download` page
- ~~adds banner prompting people to download~~ removed this cuz @ajmnz said i was showing it at the wrong place
- supersedes #2127
- fixes #2187
- learnt elixir in 2-3hrs and added this so the code is gonna be super shitty

### CODE REVIEW

File: kebab/src/raw.ts [hunk 1]
```diff
@@ -53,9 +53,9 @@ export const connect = (
   token: Token,
   refreshToken: Token,
   {
-    logger = () => {},
-    onConnectionTaken = () => {},
-    onClearTokens = () => {},
+    logger = () => { },
+    onConnectionTaken = () => { },
+    onClearTokens = () => { },
```
Comment (BronzW) [inline kebab/src/raw.ts]: Turn off formatting on save in VSCode

Comment (amitojsingh366) [reply kebab/src/raw.ts]: vro i like auto format. but i dont like this, tell me how to turn only this off

File: kibbeh/public/locales/en/translation.json [hunk 2]
```diff
@@ -294,4 +299,4 @@
       "spottedLife": "Planet with life spotted"
     }
   }
-}
+}
```
Comment (BronzW) [inline kibbeh/public/locales/en/translation.json:308]: No newline

File: kousa/env.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export TWITTER_API_KEY=
+export TWITTER_SECRET_KEY=
+export TWITTER_BEARER_TOKEN=
+export GITHUB_CLIENT_SECRET=
+export SENTRY_DNS=
+export API_URL=http://localhost:4001
+export WEB_URL=http://localhost:3000
+export PORT=4001
+export DISCORD_CLIENT_ID=829620188831678474
+export DISCORD_CLIENT_SECRET=wOM9loTVDNOStJsaIH51mr64BIBJCBSp
```
Comment (BronzW) [inline kousa/env.txt]: lmao

Comment (BronzW) [reply kousa/env.txt]: reset ur discord

Comment (amitojsingh366) [reply kousa/env.txt]: oh no, well nvm it dont matter cuz its just for testang

File: kibbeh/public/locales/en/translation.json (hunk 2 above)
Comment (amitojsingh366) [reply kibbeh/public/locales/en/translation.json:308]: ~~bero fuks u, atleast give proper reviews~~ i thought u were trolling, anyway i would like to know how to stop vsc from adding new line also

File: kousa/lib/onion/room_session.ex [hunk 4]
```diff
@@ -170,23 +174,29 @@ defmodule Onion.RoomSession do
...
   end
 
-  def add_speaker(room_id, user_id, muted?) when is_boolean(muted?) do
-    cast(room_id, {:add_speaker, user_id, muted?})
+  def add_speaker(room_id, user_id, muted?, deafened?) when is_boolean(muted?) and is_boolean(deafened?) do
+    cast(room_id, {:add_speaker, user_id, muted?, deafened?})
   end
 
-  def add_speaker_impl(user_id, muted?, state) do
+  def add_speaker_impl(user_id, muted?, deafened? ,state) do
```
Comment (BronzW) [inline kousa/lib/onion/room_session.ex]: space between ? and ,

**Review by BronzW — CHANGES_REQUESTED** [review]
Not ready for merge

File: kebab/src/raw.ts (hunk 1 above)
Comment (BronzW) [reply kebab/src/raw.ts]: VSCode settings.json `"editor.formatOnSave": false`

File: kibbeh/public/locales/en/translation.json (hunk 2 above)
Comment (BronzW) [reply kibbeh/public/locales/en/translation.json:308]: ?

**Comment by amitojsingh366** [comment]
> Not ready for merge

dude stop trolling.

**Comment by overlisted** [comment]
he's not trolling???

his review is completely reasonable

**Comment by amitojsingh366** [comment]
> he's not trolling???
> 
> his review is completely reasonable

fine, i dont wanna argue. ill just google how to fix this. anyway this is mergeable

**Comment by amitojsingh366** [comment]
b

> > he's not trolling???
> > his review is completely reasonable
> 
> fine, i dont wanna argue. ill just google how to fix this. anyway this is mergeable

**Comment by TheOtterlord** [comment]
> note to merger: use squash and merge to remove the secret from history (ik it's leaked already, but it's good to remove it from commit history)

**Comment by amitojsingh366** [comment]
> > note to merger: use squash and merge to remove the secret from history (ik it's leaked already, but it's good to remove it from commit history)

i revoked all those and also those were only for testing so its not a prob

**Comment by BronzW** [comment]
Resolves #1564

**Comment by BronzW** [comment]
Resolves #2043

File: kebab/src/raw.ts [hunk 5]
```diff
@@ -80,9 +81,8 @@ export const connect = (
       WebSocket,
     });
     const apiSend = (opcode: Opcode, data: unknown, fetchId?: FetchID) => {
-      const raw = `{"op":"${opcode}","d":${JSON.stringify(data)}${
-        fetchId ? `,"fetchId":"${fetchId}"` : ""
-      }}`;
+      const raw = `{"op":"${opcode}","d":${JSON.stringify(data)}${fetchId ? `,"fetchId":"${fetchId}"` : ""
+        }}`;
```
Comment (overlisted) [inline kebab/src/raw.ts:85]: fix your auto formatter

Comment (amitojsingh366) [reply kebab/src/raw.ts]: what's wrong here?

Comment (overlisted) [reply kebab/src/raw.ts:85]: your vscode isn't using prettierjs

File: kebab/src/raw.ts (hunk 1 above)
Comment (amitojsingh366) [reply kebab/src/raw.ts]: @BronzW i have that off. i format manually. i want to know how to stop it from adding spaces to `{}`

File: kebab/src/raw.ts (hunk 5 above)
Comment (amitojsingh366) [reply kebab/src/raw.ts]: oh i see, wtf

Comment (amitojsingh366) [reply kebab/src/raw.ts]: > your vscode isn't using prettierjs

wait, i did install that extension ben told

File: kibbeh/public/locales/en/translation.json (hunk 2 above)
Comment (amitojsingh366) [reply kibbeh/public/locales/en/translation.json:308]: do we need a newline here?

**Comment by benawad** [comment]
3 prs plz, one for each issue fixed

**Comment by amitojsingh366** [comment]
> 3 prs plz, one for each issue fixed

oh my god. ok

**Comment by amitojsingh366** [comment]
closed in favor of #2200

**Pull request closed without merging**